recipes/hindu.recipe

```diff
@@ -30,10 +30,26 @@ class TheHindu(BasicNewsRecipe):
     ]
 
     def preprocess_html(self, soup):
-        for img in soup.findAll('img', attrs={'data-src-template': True}):
-            img['src'] = img['data-src-template'].replace('BINARY/thumbnail', 'alternates/FREE_660')
+        img = soup.find('img', attrs={'class': 'lead-img'})
+        try:
+            src = img.parent.find('source').get('srcset')
+            img['src'] = re.sub(r'(ALTERNATES)/.+?/', r'\1/FREE_660/', src)
+        except (TypeError, AttributeError):
+            pass
+        # Remove duplicate intro
+        for h in soup.findAll('h2', attrs={'class': 'intro'})[1:]:
+            h.extract()
         return soup
 
+    def populate_article_metadata(self, article, soup, first):
+        try:
+            desc = soup.find('meta', attrs={'name': 'description'}).get('content')
+            if not desc.startswith('Todays paper'):
+                desc += '...' if len(desc) >= 199 else ''   # indicate truncation
+                article.text_summary = article.summary = desc
```

Comment:
One last thing, `article.text_summary = clean_ascii_chars(desc)`
needed?

Reply:
It's safer, but not strictly needed.